modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/osb-site-initializer-customer-portal-test/src/testFunctional/tests/CP2KeyDetailsDXP.testcase

```diff
@@ -0,0 +1,273 @@
+		}
+
+		task ("When a user selects a project and an individual key") {
+			CustomerPortal.enterOnProject(title = "2022 FIFA World Cup Qatar - FAN ID");
+
+			CP2NavigationMenu.goToProductActivation();
+
+			CP2NavigationMenu.clickOnProductButton(product = "DXP");
+
+			CP2ActivationKeys.viewActivationKeyDetails(key_environmentName = "Redacted");
+		}
+
+		task ("Then the modal will show all elements of the Environment group") {
+			CP2ActivationKeys.assertKeyDetailsEnvironmentElements(
+				environmentType = "Backup",
+				instanceSize = "1",
+				product = "DXP",
+				subscriptionType = "Subscription",
+				version = "7.0");
+		}
+	}
+
+	@description = "LPS-148084 - Verify that an On-Premise activation key's details pop up has all necessary information of the Server group."
+	@priority = "4"
+	test IfServerElementsArePresent {
+		property portal.upstream = "quarantine";
+
+		// Needs refactor since it depends on example FIFA account
+
+		task ("Given a user signs in on OKTA to update Customer Portal") {
+			CustomerPortal.loginOktaToUpdateCP();
+
+			Navigator.openURL();
+
+			ApplicationsMenu.gotoSite(key_site = "${testSiteName}");
+		}
+
+		task ("When a user selects a project and an individual key with an On-Premise key type") {
+			CustomerPortal.enterOnProject(title = "2022 FIFA World Cup Qatar - FAN ID");
+
+			CP2NavigationMenu.goToProductActivation();
+
+			CP2NavigationMenu.clickOnProductButton(product = "DXP");
+
+			CP2ActivationKeys.viewActivationKeyDetails(key_environmentName = "Redacted");
+		}
+
+		task ("Then the modal will show the 'Host Name', 'IP Addresses', and 'MAC Addresses' on the Server group") {
+			CP2ActivationKeys.assertKeyDetailsServerElements(
+				hostName = "Redacted",
+				ipAddress = "Redacted",
+				keyType = "On-Premise",
+				macAddress = "Redacted");
+		}
+	}
+
+	@description = "LPS-148087 - Verify that an activation key's details pop up has all necessary information of the Activation Status group."
+	@priority = "4"
+	test IfStatusElementsArePresent {
+		property portal.upstream = "quarantine";
+
+		// Needs refactor since it depends on example FIFA account
+
+		task ("Given a user signs in on OKTA to update Customer Portal") {
+			CustomerPortal.loginOktaToUpdateCP();
+
+			Navigator.openURL();
+
+			ApplicationsMenu.gotoSite(key_site = "${testSiteName}");
+		}
+
+		task ("When a user selects a project and an individual key") {
+			CustomerPortal.enterOnProject(title = "2022 FIFA World Cup Qatar - FAN ID");
+
+			CP2NavigationMenu.goToProductActivation();
+
+			CP2NavigationMenu.clickOnProductButton(product = "DXP");
+
+			CP2ActivationKeys.viewActivationKeyDetails(key_environmentName = "Redacted");
+		}
+
+		task ("Then the modal will show all elements of the Activation Status group") {
+			CP2ActivationKeys.assertKeyDetailsStatusElements(
+				expirationDate = "Mar 30, 2023",
+				startDate = "Jun 30, 2021",
+				status = "Activated");
+		}
+	}
```

Comment:
Do you think this test can be combined with IfEnvironmentElementsArePresent? They look pretty much the same with the exception of the last assertion.